File: testsrc/net/jimmc/util/TestStringUtil.java
package net.jimmc.util;

import junit.framework.Test;
import junit.framework.TestCase;

public class TestStringUtil extends TestCase {
    public TestStringUtil(String name) {
    	super(name);
    }

    public static Test suite() {
	Class testClass = TestStringUtil.class;
	Class[] coverageClasses = { StringUtil.class };
	return SuiteHelper.testOrCoverage(testClass,coverageClasses);
    }

    public void testEqual() {
        assertTrue(StringUtil.equals(null,null));
        assertTrue(StringUtil.equals("",""));
        assertTrue(StringUtil.equals("a","a"));
        assertFalse(StringUtil.equals("a",null));
        assertFalse(StringUtil.equals("a",""));
        assertFalse(StringUtil.equals(null,"b"));
        assertFalse(StringUtil.equals("","b"));
        assertFalse(StringUtil.equals(null,""));
        assertFalse(StringUtil.equals("",null));
        assertFalse(StringUtil.equals("a","b"));
        assertFalse(StringUtil.equals("a","ab"));
    }
}

File: testsrc/net/jimmc/util/TestStringUtil.scala
package net.jimmc.util

import org.scalatest.Suite

class TestStringUtil extends Suite {

    def testLongestLineWidth() {
        val s0 = null
        val n0 = StringUtil.getLongestLineWidth(s0)
        assert(0===n0)

        val s1 = "abc"
        val n1 = StringUtil.getLongestLineWidth(s1)
        assert(3===n1)

        val s2 = "a\nbc\ndef\ngh\ni\n"
        val n2 = StringUtil.getLongestLineWidth(s2)
        assert(3===n2)

        val s3 = "a\nbc\ndefg"
        val n3 = StringUtil.getLongestLineWidth(s3)
        assert(4===n3)

    }
}
